### PR TITLE
add Wprime_Wh_WhadhVV2l2nu and Zprime_Zh_ZhadhVV2l2nu cards

### PR DESCRIPTION
**Pull request by clelange**
Replaces #1778 by @cgalloni

Card generation scripts are adjusted copies of https://github.com/cms-sw/genproductions/blob/master/bin/MadGraph5_aMCatNLO/cards/production/13TeV/exo_diboson/Spin-1/Wprime_VBF_WZ_inclu/Wprime_VBF_WZ_inclu_narrow.sh

Example output is given for 2 TeV mass points.

@aspiezia @zucchett

### CODE REVIEW

File: bin/MadGraph5_aMCatNLO/cards/production/13TeV/exo_diboson/Spin-1/Wprime_Wh_WhadhVV2l2nu/Wprime_Wh_WhadhVV2l2nu_narrow_M/Wprime_Wh_WhadhVV2l2nu_narrow_M_run_card.dat [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+#*********************************************************************
+# Beam polarization from -100 (left-handed) to 100 (right-handed)    *
+#*********************************************************************
+        0     = polbeam1 ! beam polarization for beam 1
+        0     = polbeam2 ! beam polarization for beam 2
+#*********************************************************************
+# PDF CHOICE: this automatically fixes also alpha_s and its evol.    *
+#*********************************************************************
+ 'lhapdf'  = pdlabel     ! PDF set                                     
+ 263400    = lhaid     ! if pdlabel=lhapdf, this is the lhapdf number
```
Comment (perrozzi) [inline bin/MadGraph5_aMCatNLO/cards/production/13TeV/exo_diboson/Spin-1/Wprime_Wh_WhadhVV2l2nu/Wprime_Wh_WhadhVV2l2nu_narrow_M/Wprime_Wh_WhadhVV2l2nu_narrow_M_run_card.dat]: please use the PDF as described here https://twiki.cern.ch/twiki/bin/viewauth/CMS/Moriond18MC#Checks_to_the_datacards_BEFORE_T

File: bin/MadGraph5_aMCatNLO/cards/production/13TeV/exo_diboson/Spin-1/Wprime_Wh_WhadhVV2l2nu/Wprime_Wh_WhadhVV2l2nu_narrow_M2000/.dat [hunk 2]
```diff
@@ -0,0 +1 @@
+## Models for searches of diboson resonances\ndibosonResonanceModel.tar.gz
```
Comment (perrozzi) [inline bin/MadGraph5_aMCatNLO/cards/production/13TeV/exo_diboson/Spin-1/Wprime_Wh_WhadhVV2l2nu/Wprime_Wh_WhadhVV2l2nu_narrow_M2000/.dat]: this seems to be a spurious file, please remove

File: bin/MadGraph5_aMCatNLO/cards/production/13TeV/exo_diboson/Spin-1/Zprime_Zh_ZhadhVV2l2nu/Zprime_Zh_ZhadhVV2l2nu_narrow_M2000/Zprime_Zh_ZhadhVV2l2nu_narrow_M2000_run_card.dat [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+ 0.4  =  dparameter 
+#*********************************************************************
+# maximal pdg code for quark to be considered as a light jet         *
+# (otherwise b cuts are applied)                                     *
+#*********************************************************************
+ 4 = maxjetflavor    ! Maximum jet pdg code
+#*********************************************************************
+# Jet measure cuts                                                   *
+#*********************************************************************
+ 0   = xqcut   ! minimum kt jet measure between partons
```
Comment (perrozzi) [inline bin/MadGraph5_aMCatNLO/cards/production/13TeV/exo_diboson/Spin-1/Zprime_Zh_ZhadhVV2l2nu/Zprime_Zh_ZhadhVV2l2nu_narrow_M2000/Zprime_Zh_ZhadhVV2l2nu_narrow_M2000_run_card.dat]: this looks a bit too low and can lead to the infamous hanging of mg5_amc when the phase space is too inclusive. which is the qcut used in the pythia cfg?

**Comment by efeyazgan** [comment]
Hi @clelange , any follow up on this? thanks.

File: bin/MadGraph5_aMCatNLO/cards/production/13TeV/exo_diboson/Spin-1/Wprime_Wh_WhadhVV2l2nu/Wprime_Wh_WhadhVV2l2nu_narrow_M/Wprime_Wh_WhadhVV2l2nu_narrow_M_run_card.dat (hunk 1 above)
Comment (clelange) [reply bin/MadGraph5_aMCatNLO/cards/production/13TeV/exo_diboson/Spin-1/Wprime_Wh_WhadhVV2l2nu/Wprime_Wh_WhadhVV2l2nu_narrow_M/Wprime_Wh_WhadhVV2l2nu_narrow_M_run_card.dat]: I replaces line 52 with the two lines listed on the twiki. Is this correct?

File: bin/MadGraph5_aMCatNLO/cards/production/13TeV/exo_diboson/Spin-1/Wprime_Wh_WhadhVV2l2nu/Wprime_Wh_WhadhVV2l2nu_narrow_M2000/.dat (hunk 2 above)
Comment (clelange) [reply bin/MadGraph5_aMCatNLO/cards/production/13TeV/exo_diboson/Spin-1/Wprime_Wh_WhadhVV2l2nu/Wprime_Wh_WhadhVV2l2nu_narrow_M2000/.dat]: File removed.

File: bin/MadGraph5_aMCatNLO/cards/production/13TeV/exo_diboson/Spin-1/Zprime_Zh_ZhadhVV2l2nu/Zprime_Zh_ZhadhVV2l2nu_narrow_M2000/Zprime_Zh_ZhadhVV2l2nu_narrow_M2000_run_card.dat (hunk 3 above)
Comment (clelange) [reply bin/MadGraph5_aMCatNLO/cards/production/13TeV/exo_diboson/Spin-1/Zprime_Zh_ZhadhVV2l2nu/Zprime_Zh_ZhadhVV2l2nu_narrow_M2000/Zprime_Zh_ZhadhVV2l2nu_narrow_M2000_run_card.dat]: Shall I just remove this line? The default hadroniser is used, I don't think it requires a specific xqcut.

**Comment by clelange** [comment]
I'm not sure I fully understand all the comments. Please let me know if you require further changes.
About the PDF lines: should I replace the lines for M2000 in the same way, i.e.:

```
$DEFAULT_PDF_SETS = lhaid
$DEFAULT_PDF_MEMBERS = reweight_PDF
```

or leave them as they are?

**Comment by agrohsje** [comment]
Hi @clelange . I am cleaning up our PR's. This one is very old. Please let me know what you like to do with it. Should I close or should we finish?

**Comment by clelange** [comment]
@agrohsje There are signal samples that have been produced with these cards, so I guess this is a question of completeness. However, I'm not sure if there are later versions of these cards somewhere in genproductions. There are, however, similar cards with different final states in genproductions, so maybe it's not crucial to have these here. I leave it up to you. I can make smaller changes if necessary.

**Comment by agrohsje** [comment]
Hi @clelange, let's finish this. I have only a small correction. You do define q = u c d s b u~ c~ d~ s~ b~ ? Is the model per se 4FS or 5FS. You don't define the p, but use maxjetflavor 4 and  5  = asrwgtflavor?

**Comment by clelange** [comment]
Hi @agrohsje - I think this model would in principle also support 5FS, but studies in the past have shown that it has a negligible effect if one uses 4FS vs. 5 FS. I understand that the default PDFs use 5FS, so I can change it in any way you like. If you indicate what you would like me to change in the code, I can take care of it.

**Comment by agrohsje** [comment]
Hi Clemens. 
If the cards are for 2017/2018 all PDF lines should be replaced by 
$DEFAULT_PDF_SETS = lhaid
$DEFAULT_PDF_MEMBERS = reweight_PDF
I also fixed the customize file (dibosonResonanceModel.tar.gz containts the UFO you like to load, there was a mixture, I first corrected it in the wrong direction, now it should be ok.)  
Now everything is 4FS.
Let me know in case you disagree with my modifications.
Thanks. 

@sihyunjeon Do you mind x-checking as well. Thanks!

**Comment by clelange** [comment]
Sorry about the delay. Originally, these cards were for 2016. I could create a second shell script that replaces the PDF lines, but having started that I just found that all run cards already contain:

```
 $DEFAULT_PDF_SETS = lhaid
 $DEFAULT_PDF_MEMBERS = reweight_PDF
```

Is there then anything else that you would like me to do? The other modification you made look reasonable to me.

File: bin/MadGraph5_aMCatNLO/cards/production/13TeV/exo_diboson/Spin-1/Wprime_Wh_WhadhVV2l2nu/Wprime_Wh_WhadhVV2l2nu_narrow_M/Wprime_Wh_WhadhVV2l2nu_narrow_M_proc_card.dat [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import model Vector_Triplet_free_Width_UFO
+define q = u c d s b u~ c~ d~ s~ b~
+define vc = vc+ vc-
+define vw = w+ w-
+define lep+ = e+ mu+ ta+
+define lep- = e- mu- ta-
+generate p p > vc, (vc > h vw, vw > q q, (h > z > lep+ lep- vl vl~ / vz vc ta+ ta-))
+add process p p > vc, (vc > h vw, z > q q, (h > vw > lep+ lep- vl vl~ / vz vc ta+ ta-))
```
Comment (sihyunjeon) [inline bin/MadGraph5_aMCatNLO/cards/production/13TeV/exo_diboson/Spin-1/Wprime_Wh_WhadhVV2l2nu/Wprime_Wh_WhadhVV2l2nu_narrow_M/Wprime_Wh_WhadhVV2l2nu_narrow_M_proc_card.dat:8]: @agrohsje Sorry for the late review.
@clelange Can you double check this part? there's no Z in this process but you decay Z to q q. Perhaps a typo.

Comment (clelange) [reply bin/MadGraph5_aMCatNLO/cards/production/13TeV/exo_diboson/Spin-1/Wprime_Wh_WhadhVV2l2nu/Wprime_Wh_WhadhVV2l2nu_narrow_M/Wprime_Wh_WhadhVV2l2nu_narrow_M_proc_card.dat]: Good catch, I've fixed it!

**Comment by sihyunjeon** [comment]
@agrohsje I am now fine with this pull request

**Pull request merged**